float32[] state
float32[] action
---
float32[] next_state